### DISCUSSION TITLE
일정영역내에서 벗어난 플레이어의 ui만 끄고싶습니다.

### DISCUSSION THREAD

**Discussion by jiwoo50** [naverz/zepeto-studio-kor#1003] [5. Q&A]
월드내에 collider가 있는 영역이 있고 여기에 플레이어가 들어온 경우, 현재 영역내에 있는 플레이어의 수를 띄우는 ui를 구현하고 싶습니다. OnTriggerEnter와 OnTriggerExit을 이용해서 플레이어의 출입을 확인하고 현재 영역내의 플레이어의 수를 카운트하고 UI의 text로 현재 플레이어 수를 출력하고 있는데 다수의 플레이어가 들어오고 한명의 플레이어가 나가면 영역안에 있는 플레이어의 ui가 사라지는 문제가 있습니다. 
여기서 ui를 플레이어마다 따로 컨트롤 할수있는 방법이 있을까요?

아래 코드는 영역에서 플레이어의 수를 count하고 text에 현재 플레이어의 수를 띄우는 코드입니다.
![image](https://user-images.githubusercontent.com/72139485/178683371-0b9f1e1f-a8fe-46d5-8ce3-76e4b13839f9.png)
text변수에 현재 플레이어수를 출력하고 나가면 text변수에 ""를 할당해서 안보이도록 구현하였습니다.

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요. 플레이어의 수를 카운트하는 영역 스크립트와 UI 컨트롤러 스크립트를 별도로 분리해서 만드시는 것이 좋을 것 같습니다.
플레이어마다 보여지는 UI는 상호작용 예제(https://github.com/naverz/zepeto-world-sample/tree/main/Assets/Chapter3) 처럼 특정 영역에 왔을때만 보여지는 형태로 클라이언트 스크립트로 구현하시면 가능합니다.
